org-roam-capture.el

```diff
@@ -277,25 +277,54 @@ This function is used solely in Org-roam's capture templates: see
     (widen)
     (goto-char (point-max))))
 
+(define-error 'org-roam-malformed-template
+  "Malformed template in `org-roam-capture-templates'"
+  'org-roam-errors)
+
+(defun org-roam-capture--validate-template (template)
+  "Validate TEMPLATE by checking its elements.
+Return t if the TEMPLATE is well-formed, nil otherwise.
+As a special case, return 'group if TEMPLATE is a template-group.
+See `org-roam-capture-templates' for details."
+  (pcase template
+    (`(,(pred stringp) ,(pred stringp))
+     'group)
+    ((pred (lambda (x)
+             (>= (length x) 5)))
+     (pcase (cl-subseq template 0 5)
+       (`(,(pred stringp)
+          ,(pred stringp)
+          ,(pred symbolp)
+          ,(pred listp)
+          ,(pred stringp))
+        t)
+       (malformed-template
+        (signal 'org-roam-malformed-template
+                `((list stringp stringp symbolp listp stringp …)
+                  (list ,@malformed-template …))))))
+    (_ (user-error "Malformed template in `org-roam-capture-templates'"))))
```

Comment:
If we don't go for a new type of error, we can just `catch` and `throw` to here.

Reply:
think this is okay, it's a bit more explicit, but it seems that makes `user-error` message inaccurate, since malformed templates should signal the new error